kernel/kernel-api/src/main/java/io/delta/kernel/Snapshot.java

```diff
@@ -29,13 +29,29 @@
 public interface Snapshot {
 
   /**
-   * Get the version of this snapshot in the table.
+   * Get the version of this snapshot in the table. Returns -1 if the table is empty and so has no
+   * commits.
    *
    * @param engine {@link Engine} instance to use in Delta Kernel.
    * @return version of this snapshot in the Delta table
    */
   long getVersion(Engine engine);
 
+  /**
+   * Get the timestamp (in milliseconds since the Unix epoch) of the latest commit of this snapshot.
+   * Returns -1 if the table is empty and so has no commits.
+   *
+   * <p>When InCommitTimestampTableFeature is enabled, the timestamp is retrieved from the
+   * CommitInfo of the latest commit which can result in an IO operation.
```

Comment:
nit: i'm not sure "latest commit" is the right term here? do we see it as the "latest commit in this snapshot"?